src/clj/fluree/db/query/fql/parse.cljc

```diff
@@ -567,21 +567,30 @@
     q))
 
 (defn parse-analytical-query
-  [q]
-  (let [context       (context/extract q)
-        [vars values] (parse-values q context)
-        where         (parse-where q vars context)
-        grouping      (parse-grouping q)
-        ordering      (parse-ordering q)]
-    (-> q
-        (assoc :context context
-          :where where)
-        (cond-> (seq values) (assoc :values values)
-                grouping (assoc :group-by grouping)
-                ordering (assoc :order-by ordering))
-        parse-having
-        (parse-select context)
-        parse-fuel)))
+  ([q] (parse-analytical-query q nil))
+  ([q parent-context]
+   (let [context   (cond->> (context/extract q)
+                            parent-context (merge parent-context))
+         [vars values] (parse-values q context)
+         where     (parse-where q vars context)
+         grouping  (parse-grouping q)
+         ordering  (parse-ordering q)]
+     (-> q
+         (assoc :context context
+                :where where)
+         (cond-> (seq values) (assoc :values values)
+                 grouping (assoc :group-by grouping)
+                 ordering (assoc :order-by ordering))
+         parse-having
+         (parse-select context)
+         parse-fuel))))
+
+(defmethod parse-pattern :query
+  [[_ sub-query] _vars context]
+  (let [sub-query* (-> sub-query
+                       syntax/coerce-query
+                       (parse-analytical-query context))]
+    [(where/->pattern :query sub-query*)]))
```

Comment:
This probably doesn't matter, but I don't think this needs to be a `where` pattern because subqueries are executed by different machinery. It is not explicitly executed by the where-clause processing and doesn't have a corresponding `match-pattern` method in the `where` namespace like all other items created by `where/->pattern`.